flask_sqlalchemy/model.py

```diff
@@ -73,18 +73,31 @@ def __table_cls__(cls, *args, **kwargs):
         If no primary key is found, that indicates single-table inheritance,
         so no table will be created and ``__tablename__`` will be unset.
         """
+        # check if a table with this name already exists
+        # allows reflected tables to be applied to model by name
         key = _get_table_key(args[0], kwargs.get('schema'))
 
         if key in cls.metadata.tables:
             return sa.Table(*args, **kwargs)
 
+        # if a primary key or constraint is found, create a table for
+        # joined-table inheritance
         for arg in args:
             if (
                 (isinstance(arg, sa.Column) and arg.primary_key)
                 or isinstance(arg, sa.PrimaryKeyConstraint)
             ):
                 return sa.Table(*args, **kwargs)
 
+        # if no base classes define a table, return one
+        # ensures the correct error shows up when missing a primary key
+        for base in cls.__mro__[1:-1]:
+            if '__table__' in base.__dict__:
```

Comment:
I'd probably go for

```python
if not any('__table__' in b.__dict__ for b in cls.__mro__[1:-1]):
    return sa.Table(*args, **kwargs)
```

but IMO both are pretty well-readable so no strong opinon ;)

Reply:
Was doing that initially for the check above this, but it looked too complicated. Kept it the same here for consistency.